### PR TITLE
Issue 3620 fix

### PR DESCRIPTION
**Pull request by tombriles**
This pull request Fixes: #3620

### CODE REVIEW

**Comment by muxator** [comment]
Thank you very much!

Commit 891f000e1e8e69ca9abf7c51f9a9eb6fe71290ef (Test for Issue-3620: space in the HTML for an unordered list) is hard to follow: probably the introduction of the new test was intermixed with reformatting & reorganization of the code.

It should be split so it can be audited, thanks.

**Comment by tombriles** [comment]
You're asking me to remove the formerly 'pending' tests that are now passing?

**Comment by muxator** [comment]
Hi @tombriles, the fact that you took care of updating the tests before fixing the issue is commendable, thanks for that.

The thing is, before merging a PR I inspect the changes and try to assess how it changes its behaviour. But when looking at a big patch, both a trivial formatting change, a smart refactoring, or a new test that shows a failure to be fixed in the next commit, all appear the same. This makes it difficult to tell what is the intent of the change, and ensure no other bugs are introduced.

In the case of 891f000e1e8e69ca9abf7c51f9a9eb6fe71290ef, it is difficult to tell apart the syntactic changes (reformattings, whitespace changes, added trailing semicolons) from the "real stuff" (the valuable work that was done on the tests).

Some examples of purely syntactical changes that, if are to be maintained (which is fine), should go in dedicated commits:
- added spaces or semicolons:
  [here](https://github.com/ether/etherpad-lite/pull/3622/commits/891f000e1e8e69ca9abf7c51f9a9eb6fe71290ef#diff-59cfbc66b5f9988c447ac566f93ab93eL93):
  ```diff
  - var htmlWithBullets = '<html><body><ul class="list-bullet1"><li>bullet line 1</li>...</html>'
  + var htmlWithBullets = '<html><body><ul class="list-bullet1"> <li>bullet line 1</li>...</html>'
  ```
  or [here](https://github.com/ether/etherpad-lite/pull/3622/commits/891f000e1e8e69ca9abf7c51f9a9eb6fe71290ef#diff-59cfbc66b5f9988c447ac566f93ab93eL111):
  ```diff
  - importrequest(htmlWithBullets,importurl,"html")
  + importrequest(htmlWithBullets,importurl,"html");
  ```
- [reformattings](https://github.com/ether/etherpad-lite/pull/3622/commits/891f000e1e8e69ca9abf7c51f9a9eb6fe71290ef#diff-59cfbc66b5f9988c447ac566f93ab93eL96):
  ```diff
  -      [...] expect(getinnertext()).to.be('\
  -<ul class="list-bullet1"><li><span class="">bullet line 1</span></li></ul>\n\
  +      [...] expect(getinnertext()).to.be(
  +        '<ul class="list-bullet1"><li><span class="">bullet line 1</span></li></ul>\n' +
  ```

What is left after that is the real work I would lke to see: was the code reorganized? Was a new test added? Was at some point replaced `xit()` with `it()`?

I am sure there is a reason for each of these actions. I simply need them to be explained by the commits.

Thanks.

**Comment by muxator** [comment]
I cannot pull in this PR in its current state, but I'll leave it open since the fix is useful.

**Comment by JohnMcLear** [comment]
@muxator you did the right thing here.  I will try bisect the fix and send through a cleaner PR.

**Comment by JohnMcLear** [comment]
Right I read through all of this and actually it can be merged.  While I agree it's not ideal to merge so much code after doing a decent review it's all of the same fixes I would have put in.  

Obviously in the future we would prefer if @tombriles can ensure each major change is a separate commit.

I'm merging.  Again @muxator you did the right thing in pushing back and we ask @tombriles to provide smaller commits in the future.  Still, awesome work guys and thanks :D

**Comment by muxator** [comment]
These changes caused a very bad problem with spaces and the cursor jumping around with both Firefox and Chrome (see #3728).

I had to revert it via #3729.

File: src/static/js/contentcollector.js [hunk 1]
```diff
@@ -406,12 +406,13 @@ function makeContentCollector(collectStyles, abrowser, apool, domInterface, clas
...
-        node:node,
-        text:txt,
+        node: node,
+        text: txt,
         styl: null,
         cls: null
       });
-      var txt = (typeof(txtFromHook)=='object'&&txtFromHook.length==0)?dom.nodeValue(node):txtFromHook[0];
+      var txt = (typeof(txtFromHook)=='object' && txtFromHook.length==0)
+        ? dom.nodeValue(node).trim() : txtFromHook[0].trim();
```
Comment (JohnMcLear) [inline src/static/js/contentcollector.js]: .trim() being the cause of the fault / error here :(
